File: alphabetSubsequence/alphabet_subsequence.py

def alphabet_subsequence(s: str) -> bool:
    # chr(ord(char)+1) if ord(char) != 122 else 'a'
    # for char in input_string
    return all(
        ord(char) < ord(s[i+1])
        for i, char in enumerate(s)
        if 0 <= (i+1) < len(s))


inputs = ['zab', 'effg', 'cdce', 'ace', 'bxz']

for inp in inputs:
    print(f'{inp} -> {alphabet_subsequence(inp)}')

File: alphabetSubSequence/alphabetSubsequence.ts
const alphabetSubsequence = (s: string): boolean => {

	const sequence = s.split('')
	const nonSequenced = sequence.find( 
		(char, idx) => sequence[idx+1] === undefined
			? undefined 
			: char.charCodeAt(0) >= sequence[idx+1].charCodeAt(0)
	);

	return nonSequenced === undefined;
}


const inputs: string[] = ['zab', 'effg', 'cdce', 'ace', 'bxz'];

console.time('alphabetSubsequence x5');

inputs.forEach( inp => {
	console.log(`${inp} -> ${alphabetSubsequence(inp)}`); })

console.timeEnd('alphabetSubsequence x5');
